### PR TITLE
Enhancement/query graph ql for claims

### LINKED ISSUE #309: [Community Site] Claim: Reduce Infura calls
There is a [piece of code](https://github.com/Taraxa-project/taraxa-platform/blob/develop/services/claim-backend/src/modules/claim/claim.service.ts#L334) in the Claim Service that does repeated Infura requests.

As of now, the data is indexed in TheGraph so we should redirect the request there.

- [ ] Write the graphQL query to fetch the same data.
- [ ] Remove the Infura request.

### PR DESCRIPTION
**Pull request by VargaElod23**
Closes #309 .

### CODE REVIEW

**Comment by VargaElod23** [comment]
Tested with input :

```JavaScript
{
	"address": "0xdd144d2b6de680c9faf1ebf1b801f85ad4b4cfe3",
	"amount": "8000000000000000000000",
	"nonce":  "9256"
}
```

File: services/claim-backend/src/modules/claim/claim.service.ts [hunk 1]
```diff
@@ -331,16 +336,20 @@ export class ClaimService {
...
         );
+        if (!indexedClaims)
+          throw new InternalServerErrorException(
+            `No such claim was indexed with nonce: ${nonce}`,
+          );
+        if (indexedClaims.length > 1)
+          throw new InternalServerErrorException(
+            `Multiple claims were indexed with nonce: ${nonce}`,
+          );
+        const confirmation = ethers.BigNumber.from(indexedClaims[0].amount);
```
Comment (rattrap) [inline services/claim-backend/src/modules/claim/claim.service.ts:352]: We shouldn't throw here. This is the account request.

File: services/claim-backend/config/reward.ts [hunk 2]
```diff
@@ -3,4 +3,5 @@ import { registerAs } from '@nestjs/config';
 export default registerAs('reward', () => ({
   communitySiteApiUrl: process.env.COMMUNITY_SITE_API_URL,
   delegationApiUrl: process.env.DELEGATION_API_URL,
+  graphQLClaimUrl: `${process.env.GRAPH_QL_CLAIM_ENDPOINT}/subgraphs/id/${process.env.CLAIM_SUBGRAPH_ID}`,
```
Comment (rattrap) [inline services/claim-backend/config/reward.ts:6]: We already have a config value for this URL in `indexer.ts`

**Review by rattrap — CHANGES_REQUESTED** [review]

File: services/claim-backend/src/modules/claim/claim.service.ts (hunk 1 above)
Comment (VargaElod23) [reply services/claim-backend/src/modules/claim/claim.service.ts:352]:
```suggestion
```

File: services/claim-backend/config/reward.ts (hunk 2 above)
Comment (VargaElod23) [reply services/claim-backend/config/reward.ts:6]: Yeah that's for the staking contract, this one is for the claim, but I'm moving this there.

Comment (rattrap) [reply services/claim-backend/config/reward.ts:6]: I mean let's also add this claim subgraph url in the indexer file.

Comment (rattrap) [reply services/claim-backend/config/reward.ts:6]: And it can include the subgraph id too

Comment (rattrap) [reply services/claim-backend/config/reward.ts:6]: Oh. Yeah. Nevermind

File: services/claim-backend/src/modules/claim/claim.module.ts [hunk 3]
```diff
@@ -30,15 +32,23 @@ import { BlockchainModule } from '@taraxa-claim/blockchain';
...
   controllers: [
     BatchController,
     RewardController,
     AccountController,
     ClaimController,
   ],
-  providers: [ClaimService],
-  exports: [ClaimService],
+  providers: [ClaimService, GraphQLService],
+  exports: [ClaimService, GraphQLService],
```
Comment (rattrap) [inline services/claim-backend/src/modules/claim/claim.module.ts:52]: No need to export it. We're using it in the same module

**Review by rattrap — APPROVED** [review]

File: services/claim-backend/src/modules/claim/claim.module.ts (hunk 3 above)
Comment (VargaElod23) [reply services/claim-backend/src/modules/claim/claim.module.ts]: Removed

**Pull request merged**